src/api/java/baritone/api/Settings.java

```diff
@@ -733,6 +733,11 @@
      */
     public final Setting<Boolean> itemSaver = new Setting<>(false);
 
+    /**
+     * Durability to leave on the tool when using itemSaver
+     */
+    public final Setting<Integer> itemSaverThreshold = new Setting<>(10);
```

Comment:
Ideally this would be one setting (-1 ≙ itemSaver off) but it's a boolean already and I don't think you should change that.
TLDR: imo you've don it right